### ISSUE TITLE
File exists error after few weeks of runtime

### ISSUE THREAD

**Issue by maurerle** [freifunkh/wireguard-vxlan-glue#10]
After a few weeks of runtime I get the following error sometimes:
```
May 31 23:11:14 wg-node02 python3[2950362]: Traceback (most recent call last):
May 31 23:11:14 wg-node02 python3[2950362]:   File "/srv/wireguard/vxlan-glue/netlink.py", line 248, in <module>
May 31 23:11:14 wg-node02 python3[2950362]:     manager.push_vxlan_configs()
May 31 23:11:14 wg-node02 python3[2950362]:   File "/srv/wireguard/vxlan-glue/netlink.py", line 139, in push_vxlan_configs
May 31 23:11:14 wg-node02 python3[2950362]:     ip.route(
May 31 23:11:14 wg-node02 python3[2950362]:   File "/usr/local/lib/python3.9/dist-packages/pyroute2/iproute/linux.py", line 2262, in route
May 31 23:11:14 wg-node02 python3[2950362]:     ret = self.nlm_request(
May 31 23:11:14 wg-node02 python3[2950362]:   File "/usr/local/lib/python3.9/dist-packages/pyroute2/netlink/nlsocket.py", line 822, in nlm_request
May 31 23:11:14 wg-node02 python3[2950362]:     return tuple(self._genlm_request(*argv, **kwarg))
May 31 23:11:14 wg-node02 python3[2950362]:   File "/usr/local/lib/python3.9/dist-packages/pyroute2/netlink/nlsocket.py", line 1129, in nlm_request
May 31 23:11:14 wg-node02 python3[2950362]:     for msg in self.get(
May 31 23:11:14 wg-node02 python3[2950362]:   File "/usr/local/lib/python3.9/dist-packages/pyroute2/netlink/nlsocket.py", line 825, in get
May 31 23:11:14 wg-node02 python3[2950362]:     return tuple(self._genlm_get(*argv, **kwarg))
May 31 23:11:14 wg-node02 python3[2950362]:   File "/usr/local/lib/python3.9/dist-packages/pyroute2/netlink/nlsocket.py", line 489, in get
May 31 23:11:14 wg-node02 python3[2950362]:     raise msg['header']['error']
May 31 23:11:14 wg-node02 python3[2950362]: pyroute2.netlink.exceptions.NetlinkError: (17, 'File exists')
```

I would suggest to add a try-catch block around `push_vxlan_configs` in the loop (or into the function), so that everything is removed accordingly, a little time is waited and the service is restarted correctly.
I am opening this issue first to see if anyone else using this repo encountered this problem.

The error happens like 7 times in a row in a short time, which results in the service shutting down and not starting again until the server is rebootet or the service is restarted manually

**Comment by maurerle** [comment]
fixed by ebbe7956c0b492b0ddd48a9d257795bbe7df14b9

but #12 is still needed

**Comment by maurerle** [comment]
The source of this seems to be, when moving a key from one segment to the other.
If the manager which removes the key comes after the manager adding the key, there already exists a route for interface X, but the other manager adds the same route to interface Y, before the first one did the removal.
This gives a File exists error.

To fix this, we should refactor

```python
for manager in managers:
    manager.pull_from_wireguard()
    manager.push_vxlan_configs()
```

to something like

```python
for manager in managers:
    manager.pull_from_wireguard()
    manager.remove_keys()

for manager in managers:
    manager.push_vxlan_configs()
```
